wpilib/wpilib/shuffleboard/widget.py

```diff
@@ -1,14 +1,21 @@
-# validated: 2018-11-18 EN 175c6c1f0130 edu/wpi/first/wpilibj/shuffleboard/ShuffleboardWidget.java
+# validated: 2019-01-05 TW 01d13220660c edu/wpi/first/wpilibj/shuffleboard/ShuffleboardWidget.java
 # ----------------------------------------------------------------------------
 # Copyright (c) 2018 FIRST. All Rights Reserved.
 # Open Source Software - may be modified and shared by FRC teams. The code
 # must be accompanied by the FIRST BSD license file in the root directory of
 # the project.
 # ----------------------------------------------------------------------------
+from typing import Union
+
 from .component import ShuffleboardComponent
 from .container import ShuffleboardContainer
 
 
+class WidgetType:
```

Comment:
Are you actually going to port the enums for this? Otherwise this is completely useless.

Reply:
Also I feel this should be an enum.Enum instead of a getter method to return its value.

Reply:
Why can't this just act as the interface class for type hints?

Reply:
I think we are again in violent agreement.

Reply:
But this works fine as is for the interface. Unless type hints has a way of specifying "idc what type it is, as long as it has these attributes"...should I change to `hasattr` within the method?

Reply:
oh I'm dumb. If they define it as one of the enums, it won't be a widget type...ugh ok

Reply:
Realistically, the only WidgetType that teams will probably care about is BuiltInWidgets. If users create their own widgets, they can pass in a string constant just fine.

Reply:
I've got a trick up my sleeve you'll probably dislike

Reply:
It'd probably be better to just have BuiltInWidgets by itself actually. More discoverable that way.

Reply:
@virtuald any strong opinions on the above?

Reply:
If there's a base widget type that is almost always going to be used, I think it'd be better to use that for now instead of introducing a new dependency.

Reply:
Looking at the C++ implementation, all WidgetTypes have a string constant name. I say my enum idea earlier is probably the way to go.